score-e2e/src/main/java/org/oagi/score/e2e/impl/page/bie/ExpressBIEPageImpl.java

```diff
@@ -31,4 +62,72 @@ public void openPage() {
     public WebElement getTitle() {
         return visibilityOfElementLocated(getDriver(), By.className("title"));
     }
+
+    @Override
+    public void selectBIEForExpression(TopLevelASBIEPObject topLevelASBIEP) {
+        ReleaseObject release = getAPIFactory().getReleaseAPI().getReleaseById(topLevelASBIEP.getReleaseId());
+        setBranch(release.getReleaseNumber());
+        setDEN(topLevelASBIEP.getDen());
+        hitSearchButton();
+
+        retry(() -> {
+            WebElement tr = getTableRecordByValue(topLevelASBIEP.getDen());
+            WebElement td = getColumnByName(tr, "select");
+            click(td.findElement(By.xpath("mat-checkbox/label/span[1]")));
+        });
+    }
+
+    @Override
+    public void setBranch(String branch) {
+        retry(() -> {
+            click(getBranchSelectField());
+            WebElement optionField = visibilityOfElementLocated(getDriver(),
+                    By.xpath("//mat-option//span[text() = \"" + branch + "\"]"));
+            click(optionField);
+        });
+    }
+
+    @Override
+    public WebElement getBranchSelectField() {
+        return visibilityOfElementLocated(getDriver(), BRANCH_SELECT_FIELD_LOCATOR);
+    }
+
+    @Override
+    public WebElement getDENField() {
+        return visibilityOfElementLocated(getDriver(), DEN_FIELD_LOCATOR);
+    }
+
+    @Override
+    public void setDEN(String den) {
+        sendKeys(getDENField(), den);
+    }
+    @Override
+    public WebElement getSearchButton() {
+        return elementToBeClickable(getDriver(), SEARCH_BUTTON_LOCATOR);
+    }
+
+    @Override
+    public void hitSearchButton() {
+        click(getSearchButton());
+        invisibilityOfLoadingContainerElement(getDriver());
+    }
+    @Override
+    public WebElement getTableRecordByValue(String value) {
+        return visibilityOfElementLocated(getDriver(), By.xpath("//td//span[contains(text(), \"" + value + "\")]/ancestor::tr"));
+    }
+    @Override
+    public WebElement getColumnByName(WebElement tableRecord, String columnName) {
+        return tableRecord.findElement(By.className("mat-column-" + columnName));
+    }
+
+    @Override
+    public void hitGenerateButton() {
+        click(getGenerateButton());
+        waitFor(Duration.ofMillis(15000));
```

Comment:
@hakjuoh I guess this is not that smart solution. The BIE generation takes some time, I suppose it should wait until the file is downloaded, not longer. Also, do we need to check if the file is successfully downloaded?

Reply:
@elena-jelisic According to [selenium docs](https://www.selenium.dev/documentation/test_practices/discouraged/file_downloads/), it recommends to find the download link using selenium and pass it to other HTTP clients, such as curl, to test file downloads, instead of delaying. Let's skip this part for now.